src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -159,7 +159,7 @@ private static string GetCorrectCasedPath(string path)
                     }
                 }
 
-                if (path.EndsWith(StringLiterals.DefaultPathSeparator))
+                if (path.EndsWith(StringLiterals.DefaultPathSeparator) && !exactPath.EndsWith(StringLiterals.DefaultPathSeparator))
                 {
                     return exactPath + StringLiterals.DefaultPathSeparator;
                 }
```

Comment:
It seems we could remove the lines at all.

Reply:
`path` is the original path that may have incorrect casing, so if it originally contained a trailing slash, we want to add it back.  In the case of the root of the drive, it already has a trailing slash so we don't want to add another one.

Reply:
> // This handles the trailing slash case

There is already a branch for this. So proposal is:
```c#
                foreach (string item in path.Split(StringLiterals.DefaultPathSeparator))
                {
                    if (itemsToSkip-- > 0)
                    {
                        // This handles the UNC server and share and 8.3 short path syntax
                        exactPath += item + StringLiterals.DefaultPathSeparator;
                        continue;
                    }
                    else if (string.IsNullOrEmpty(exactPath))
                    {
                        // This handles the drive letter or / root path start
                        exactPath = item + StringLiterals.DefaultPathSeparator;
                    }
                    else if (string.IsNullOrEmpty(item))
                    {
                        // This handles the trailing slash case
                        if (!exactPath.EndsWith(StringLiterals.DefaultPathSeparator))
                        {
                            exactPath += StringLiterals.DefaultPathSeparator;
                        }

                        break;
                    }
                    else if (item.Contains('~'))
                    {
                        // This handles short path names
                        exactPath += StringLiterals.DefaultPathSeparator + item;
                    }
                    else
                    {
                        exactPath = Directory.GetFileSystemEntries(exactPath, item).First();
                    }
                }

                return exactPath;
```

Reply:
Ok, got it.